terraform/cloudwatch.tf.json.template.py

```diff
@@ -1,34 +1,96 @@
         {
-            "aws_cloudwatch_metric_alarm": {
-                f"{lambda_}_5xx": {
-                    "alarm_name": config.qualified_resource_name(lambda_ + '_5xx'),
-                    "comparison_operator": "GreaterThanThreshold",
-                    # This alarm will catch persistent 5XX errors occurring over
-                    # one hour, specifically when more than one occurrence is
-                    # sampled in a ten-minute period for six consecutive periods.
-                    "evaluation_periods": 6,
-                    "period": 60 * 10,
-                    "metric_name": "5XXError",
-                    "namespace": "AWS/ApiGateway",
-                    "statistic": "Sum",
-                    "threshold": 1,
-                    "treat_missing_data": "notBreaching",
-                    "dimensions": {
-                        "ApiName": config.qualified_resource_name(lambda_),
-                        "Stage": config.deployment_stage,
+            "external": {
+                "elasticsearch_nodes": {
+                    "program": [
+                        "python",
+                        f"{config.project_root}/scripts/elasticsearch_nodes.py"
+                    ],
+                    "query": {},
+                    "depends_on": ([]
+                                   if config.share_es_domain else
+                                   ["aws_elasticsearch_domain.index"])
+                }
+            }
+        }
+    ],
+    "locals": {
+        "nodes": "${jsondecode(data.external.elasticsearch_nodes.result.nodes)}"
+    },
+    "resource": [
+        *(
+            {
+                "aws_cloudwatch_metric_alarm": {
+                    f"{lambda_}_5xx": {
+                        "alarm_name": config.qualified_resource_name(lambda_ + '_5xx'),
+                        "comparison_operator": "GreaterThanThreshold",
+                        # This alarm will catch persistent 5XX errors occurring over
+                        # one hour, specifically when more than one occurrence is
+                        # sampled in a ten-minute period for six consecutive periods.
+                        "evaluation_periods": 6,
+                        "period": 60 * 10,
+                        "metric_name": "5XXError",
+                        "namespace": "AWS/ApiGateway",
+                        "statistic": "Sum",
+                        "threshold": 1,
+                        "treat_missing_data": "notBreaching",
+                        "dimensions": {
+                            "ApiName": config.qualified_resource_name(lambda_),
+                            "Stage": config.deployment_stage,
+                        }
                     }
                 }
             }
+            for lambda_ in config.lambda_names()
+            if config.enable_monitoring
+        ),
+        {
+            "aws_cloudwatch_dashboard": {
+                config.qualified_resource_name("dashboard"): {
```

Comment:
Why is the TF resource name dynamic? We don't do that unless we have to, and if we do we don't use qualification.